File: samples/java-app/src/main/java/com/appambit/javaapp/SecondActivity.java
package com.appambit.javaapp;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;

public class SecondActivity extends AppCompatActivity {

    Button btnReturnToMain;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_second);

        btnReturnToMain = findViewById(R.id.btnReturnMain);
        btnReturnToMain.setOnClickListener(v -> {
            finish();
            startActivity(new Intent(getApplicationContext(), MainActivity.class));
        });

    }
}

File: appambit-kotlin-testapp/src/main/java/com/appambit/kotlintestapp/SecondActivity.kt
package com.appambit.kotlintestapp

import android.content.Intent
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material3.Button
import androidx.compose.material3.Text
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier

class SecondActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContent {
            Box(
                modifier = Modifier
                    .fillMaxSize(),
                contentAlignment = Alignment.Center
            ) {
                Button(
                    onClick = { backToMainActivity() }
                ) {
                    Text(text = "Back to MainActivity")
                }
            }
        }

    }
}

fun ComponentActivity.backToMainActivity() {
    this.startActivity(Intent(this, MainActivity::class.java))
    this.finish()
}